src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1

```diff
@@ -200,7 +200,7 @@ Rule 'Azure.AKS.AvailabilityZone' -Type 'Microsoft.ContainerService/managedClust
         # Availability zones only available on virtual machine scale sets
         if ($Assert.HasFieldValue($agentPool, 'type', 'VirtualMachineScaleSets').Result) {
             $Assert.HasFieldValue($agentPool, 'availabilityZones').
-                Reason($LocalizedData.AKSAvailabilityZone, $agentPool.name, $location, $joinedZoneString);
+                Reason($LocalizedData.AKSAvailabilityZone, $agentPool.name, $TargetObject.Location, $joinedZoneString);
```

Comment:
Wonder if we should unit test Reasons? I fixed one here but it could be a good idea to test the format is correct.

Reply:
Might not be worth it though, would clutter tests quite a but 😄

Reply:
It's something that we could do better, it can be quite helpful. I think use your judgement and add if the rule or reason is complex.

Some examples:

https://github.com/Azure/PSRule.Rules.Azure/blob/e910d5f66b1b12bd645e80dbdf9c032805ad8368/tests/PSRule.Rules.Azure.Tests/Azure.APIM.Tests.ps1#L79
https://github.com/Azure/PSRule.Rules.Azure/blob/e910d5f66b1b12bd645e80dbdf9c032805ad8368/tests/PSRule.Rules.Azure.Tests/Azure.APIM.Tests.ps1#L177

Reply:
@BernieWhite Do you mind if I create an issue for this and I can go through and find the tests that are not testing the Reason? I can probably do all that in a separate PR 🙂 .